task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawer.vue

```diff
@@ -303,9 +305,43 @@ export default {
                             this.menuActions = this.menuActions.filter(menuAction => menuAction.enabled);
                         });
                     }
+                } else if ( event.detail.name === 'noProject' ) {
+                    this.menuActions = [];
+                    this.enableDelete = event.detail.task.createdBy === eXo.env.portal.userName;
+                    this.enableClone = event.detail.task.assignee === eXo.env.portal.userName || event.detail.task.coworker.includes(eXo.env.portal.userName);
+                    this.addMenuAction(this.$t('label.delete'), 'uiIconTrash', this.enableDelete, 'deleteTask');
+                    this.addMenuAction(this.$t('label.clone'), 'uiIconCloneNode', this.enableClone, 'cloneTask');
+                    this.menuActions = this.menuActions.filter(menuAction => menuAction.enabled);
                 }
             }
         });
+       const urlPath = document.location.pathname
+       if(urlPath.includes('taskDetail')){
+         let taskId = urlPath.split('taskDetail/')[1].split(/[^0-9]/)[0]
+         taskId = taskId && Number(taskId) || 0;
+         if (taskId) {
+           this.$tasksService.getTaskById(taskId).then(data => {
+             this.task = data  
+             if(this.task.status && this.task.status.project) {
```

Comment:
This whole block is duplicated, please refactor it to avoid that.

Reply:
Done @ahamdi